packages/runtime/container-runtime/src/deltaScheduler.ts

```diff
@@ -25,81 +29,111 @@ import {
 export class DeltaScheduler {
     private readonly deltaManager: IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>;
     // The time for processing ops in a single turn.
-    public static readonly processingTime = 20;
+    public static readonly processingTime = 50;
 
     // The increase in time for processing ops after each turn.
     private readonly processingTimeIncrement = 10;
 
     private processingStartTime: number | undefined;
     private totalProcessingTime: number = DeltaScheduler.processingTime;
 
-    // This keeps track of whether the delta scheduler is scheduling a particular run of the
-    // the inbound delta queue. Basically, every time the delta queue starts processing with
-    // more than one op, this will be set to true until the run completes.
-    private isScheduling: boolean = false;
-
     // This keeps track of the number of times inbound queue has been scheduled. After a particular
     // count, we log telemetry for the number of ops processed, the time and number of turns it took
     // to process the ops.
     private schedulingCount: number = 0;
 
     private schedulingLog: {
-        numberOfOps: number;
+        opsRemainingToProcess: number;
         totalProcessingTime: number;
         numberOfTurns: number;
     } | undefined;
 
+    // Keeps track of the batching numbers so we can have an idea on how many ops were processed.
+    private batchProcessBegin: boolean = false;
+    private numberOfBatchesProcessed: number = 0;
+    private lastSequenceNumber: number = 0;
+    private firstSequenceNumber: number = 0;
+
     constructor(
         deltaManager: IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>,
         private readonly logger: ITelemetryLogger,
     ) {
         this.deltaManager = deltaManager;
         this.deltaManager.inbound.on("idle", () => { this.inboundQueueIdle(); });
+        this.deltaManager.inbound.on("op", (op: ISequencedDocumentMessage) => {
+            if (op.type === MessageType.Operation) {
```

Comment:
I don't think this is correct. For each op, this handler will probably be called before batchBegin which is called when container runtime processes the op (after delta manager). So, batchProcessBegin will be false even though this may be a batch.

Reply:
I think it would be better to add the op as a parameter to batchBegin and beginEnd. ScheduleManager can pass in the op when it calls those methods.